### ISSUE TITLE
Export portal file from bsp

### ISSUE THREAD

**Issue by mfaizsyahmi** [wootguy/bspguy#97]
A shot in the dark here, but related to #96. 

I want to re-VIS a bsp file so that the area with `info_overview_point` is visible everywhere on the map. But VIS program requires a portal file, which doesn't come with maps you didn't compile yourselves.

There is a program called bsp2prt out in the wild, but it only takes in quake bsp maps. It refuses to take in goldsrc's bsp files. But this proves that it's technically possible to fish out portal data from a compiled BSP.

If #96 isn't going to be pursued, It'd be mighty helpful to finally have the goldsrc counterpart to the program, so that we can re-VIS existing maps. Perhaps to fix VIS glitches from the original VIS with modern tools.

**Comment by Qwertyus3D** [comment]
@mfaizsyahmi I think it's probably possible to edit BSP in any HEX-editor and change its' header, to specify another BSP version. So this will hopefully 'demake' BSP into file, compatible with bsp2prt.

**Comment by marigi64** [comment]
This unfortunately doesn't seem to work with bsp2prt. Changing the header to version 29 results in the program not generating any prt file.

**Comment by mfaizsyahmi** [comment]
> @mfaizsyahmi I think it's probably possible to edit BSP in any HEX-editor and change its' header, to specify another BSP version. So this will hopefully 'demake' BSP into file, compatible with bsp2prt.

Doesn't work. I tested this route before I raised this issue.

**Comment by UnrealKaraulov** [comment]
@mfaizsyahmi bsp2ptr is opensource tool? how to download it?

I think it needed feature because some manipulation can destroy VIS data.

**Comment by UnrealKaraulov** [comment]
@wootguy you don't know, BSP file contains needed information for restoring .PRT file?

**Comment by wootguy** [comment]
I don't think it does, but marking which leaves you want visible is doable. I guess there would be a leaf selection mode where selecting a face highlights all the faces in the same leaf. Then you could have a widget that lets you set the leaf to be globally visible.

I do feel like it _should_ be possible to re-compile a bsp, starting from the BSP stage. After CSG finishes you get a set of faces which I think is similar to the final bsp but without any cuts. I couldn't get hlbsp to work with any faces I made though. I don't understand the compilation process well enough to get that working.

**Comment by UnrealKaraulov** [comment]
@wootguy I think PVS can be edited without problem, but if need do revis, vis.exe requires .prt file, i don't know file format and specs, something like

ptr file
<leaf num>
<portals num>
LEAFS:
1
1
1... 
PORTALS:
<NUM VERTS> <LEAF1> <LEAF2> <xyz num verts times>

?

**Comment by marigi64** [comment]
Being able to edit Vis or recompile it would certainly come in handy. Some faces don't display in many old maps and I've been looking for a way to fix that without having to recompile the entire map.

![20230801150748_1](https://github.com/wootguy/bspguy/assets/53112738/e567aeb7-9196-46a1-876c-461242e5861a)

**Comment by wootguy** [comment]
I unknowingly made some progress on this. The [portal file](https://developer.valvesoftware.com/wiki/PRT) is a list of polygons describing how leaves are connected. I made those as part of a navigation mesh generator I was working on. Dumping those polygons might be enough to run VIS but I doubt it's that simple.

**Comment by Rosenstein** [comment]
👍

**Comment by Rosenstein** [comment]
❤️

**Comment by wootguy** [comment]
Implemented in v6. Instructions [here](https://github.com/wootguy/bspguy/wiki/PVS-Editing#vis-recompilation).
